clang/lib/Sema/SemaOpenACC.cpp

```diff
@@ -2589,26 +2589,63 @@ SemaOpenACC::ActOnOpenACCAsteriskSizeExpr(SourceLocation AsteriskLoc) {
   return BuildOpenACCAsteriskSizeExpr(AsteriskLoc);
 }
 
-/// Loops through a type and generates an appropriate InitListExpr to generate
-/// type initialization.
-static Expr *GenerateReductionInitRecipeExpr(ASTContext &Context,
-                                             SourceRange ExprRange,
-                                             QualType Ty) {
+namespace {
+enum class InitKind { One, AllOnes, Least, Largest };
+llvm::APFloat getInitFloatValue(ASTContext &Context, InitKind IK, QualType Ty) {
+  switch (IK) {
+  case InitKind::One:
+    return llvm::APFloat::getOne(Context.getFloatTypeSemantics(Ty));
+  case InitKind::AllOnes:
+    return llvm::APFloat::getAllOnesValue(Context.getFloatTypeSemantics(Ty));
+  case InitKind::Least:
+    return llvm::APFloat::getSmallestNormalized(
```

Comment:
Are you sure you don't want `llvm::APFloat::getLargest(Context.getFloatTypeSemantics(Ty), /*Negative=*/true);` here? I think `getSmallestNormalized` is going to return a value near zero.

Reply:
Ah!  Good to know, thank you!